### ISSUE TITLE
Is it possible to modify a script after loading it?

### ISSUE THREAD

**Issue by red8888** [jenkinsci/JenkinsPipelineUnit#407]
After loading a script like this: 
`def script = loadScript("testScript")`

Is it possible to modify it before running tests? I'd like to have a base script file for running tests and then just swap out some parameters before running a specific test.

My Jenkinsfiles just call pipelines defined in my shared library and pass them a list of params like this:
```
@Library('shared-lib') _
pipelineScript ([<MAP OF PARAMS])
```

And then pipelineScript.groovy looks like this:
`def call(Map config) {...}`

After running loadScript() I can't figure out how to modify the `config` Map that is read in, maybe this isn't possible?

Thanks!

**Comment by nre-ableton** [comment]
Yes, you should be able to modify fields owned by the script after loading it. We have several unit tests that do stuff like this:

```groovy
  @Override
  @Before
  void setUp() {
    super.setUp()
    this.script = loadScript('test/resources/EmptyPipeline.groovy')
    script.env = [FOO: 'bar']
  }
```

However, I'm not sure to what extent you can modify the script. We typically only mess with `env` and `params`.
